cmd/ddev/cmd/composer-create.go

```diff
@@ -241,3 +240,61 @@ func init() {
 	ComposerCmd.AddCommand(ComposerCreateProjectCmd)
 	ComposerCmd.AddCommand(ComposerCreateCmd)
 }
+
+// removeAll removes all files and folders in path except the ones listed in
+// skipRecursive and skip where skip only preserves the folder itself but still
+// removes containing files and folders.
+func removeAll(aPath string, skipRecursive, skip []string) error {
+	fsObjects, err := fileutil.ListFilesInDir(aPath)
+	if err != nil {
+		return err
+	}
+
+ObjectsLoop:
+	for _, fsObject := range fsObjects {
+		// Just skip objects listed in skipRecursive.
+		for _, skipObject := range skipRecursive {
+			if fsObject == skipObject {
+				continue ObjectsLoop
+			}
+		}
+
+		for _, skipObject := range skip {
+			if strings.HasPrefix(skipObject, fsObject) {
+				// Build new skipRecursive removing current object prefix.
+				newSkipRecursive := []string{}
+				for _, skipObject := range skipRecursive {
+					if strings.HasPrefix(skipObject, fsObject) && skipObject != fsObject {
+						skipObject, _ = strings.CutPrefix(skipObject, fsObject+"/")
+						newSkipRecursive = append(newSkipRecursive, skipObject)
+					}
+				}
+
+				// Build new skip removing current object prefix.
+				newSkip := []string{}
+				for _, skipObject := range skip {
+					if strings.HasPrefix(skipObject, fsObject) && skipObject != fsObject {
+						skipObject, _ = strings.CutPrefix(skipObject, fsObject+"/")
+						newSkip = append(newSkip, skipObject)
+					}
+				}
+
+				// Recursive call of removeAll to remove content of object.
+				err = removeAll(path.Join(aPath, fsObject), newSkipRecursive, newSkip)
```

Comment:
I'm not convinced that recursion is called for here. Recursion should be used only when it's absolutely appropriate.